addon/utils/selection.js

```diff
@@ -63,24 +62,27 @@ export default {
    * @param {Function} itemComparator - comparator for items
    */
   /* eslint-disable complexity */
-  range (items, selectedItems, item, rangeState, itemComparator) {
+  range (items, selectedItems, item, rangeState, itemComparator, itemKey) {
     // If an anchor isn't set, then set the anchor and exit
     const rangeAnchor = rangeState['anchor']
-    if (isNone(rangeAnchor)) {
+
+    const anchor = items.findIndex(currentItem => itemComparator(currentItem, rangeState['anchor']))
+    // If anchor is -1 then it was a anchor from a previous page that we can not find, so reset
+    if (Ember.isEmpty(rangeAnchor) || anchor === -1) {
       // Range select is always a positive selection (no deselect)
       rangeState['anchor'] = item
 
       // New anchor, clear any previous endpoint
       rangeState['endpoint'] = Ember.Object.create()
 
-      // Add the anchor to the selected items
-      selectedItems.pushObject(item)
+      // Add the anchor to the selected items if not already in it from a previous page
+      if (!selectedItems.some(selectedItem => itemComparator(selectedItem, item))) {
```

Comment:
If you are on page 2 of pagination, switch back to page 1, your previous anchors are no longer valid.
Now that you are on page 1, if you shift select on item already selected, you will just throw it into 'selectedItems'. Since anchors are no longer valid you have no check to see if this item was already selected.